paddle/fluid/operators/elementwise/mkldnn/elementwise_add_mkldnn_op.cc

```diff
@@ -65,21 +65,27 @@ class EltwiseAddMKLDNNKernel : public framework::OpKernel<T> {
           (src_x_tz.size() == 5 &&
            x->format() != (format = memory::format::ncdhw))) {
         _x.Resize(x_dims);
-        auto user_x_memory_pd = memory::primitive_desc(
-            {{src_x_tz}, memory::data_type::f32, x->format()}, mkldnn_engine);
-        auto x_memory_pd = memory::primitive_desc(
-            {{src_x_tz}, memory::data_type::f32, format}, mkldnn_engine);
-        auto size = x_memory_pd.get_size();
-        _x.mutable_data<T>(ctx.GetPlace(), size);
-        auto user_x_memory =
-            memory(user_x_memory_pd, paddle::platform::to_void_cast<T>(x_data));
-        auto x_memory = memory(x_memory_pd,
-                               paddle::platform::to_void_cast<T>(_x.data<T>()));
-
-        auto x_reorder = reorder(user_x_memory, x_memory);
+
+        mkldnn::memory::data_type in_type = platform::MKLDNNGetDataType<T>();
+        auto out_format = platform::MKLDNNFormatForSize(
+            x_dims.size(), mkldnn::memory::format::nchw);
+
+        const std::string key = platform::ReorderMKLDNNHandler::GetHash(
+            src_x_tz, x->format(), out_format, std::to_string(in_type));
+
+        platform::ReorderMKLDNNHandler handler(src_x_tz, x->type(), in_type,
+                                               dev_ctx, mkldnn_engine, key);
+
+        auto user_x_memory_p = handler.AcquireSrcMemory(
```

Comment:
why not use std::shared_ptr<>? this mkldnn memory will be stored in shared_ptr in blob, but it is not well cached or cache is cleaned, then this should be a potential core dump point.